lib/resourcemerge/apiext.go

```diff
@@ -12,18 +12,40 @@ import (
 // modified is set to true when existing had to be updated with required.
 func EnsureCustomResourceDefinitionV1(modified *bool, existing *apiextv1.CustomResourceDefinition, required apiextv1.CustomResourceDefinition) {
 	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
+	ensureCustomResourceDefinitionV1Defaults(&required)
+	ensureCustomResourceDefinitionV1CaBundle(&required, *existing)
+	if !equality.Semantic.DeepEqual(existing.Spec, required.Spec) {
+		*modified = true
+		existing.Spec = required.Spec
+	}
+}
 
-	// apply defaults to blue print
+func ensureCustomResourceDefinitionV1Defaults(required *apiextv1.CustomResourceDefinition) {
 	if len(required.Spec.Names.Singular) == 0 {
 		required.Spec.Names.Singular = strings.ToLower(required.Spec.Names.Kind)
 	}
 	if len(required.Spec.Names.ListKind) == 0 {
 		required.Spec.Names.ListKind = required.Spec.Names.Kind + "List"
 	}
+}
 
-	// we stomp everything
-	if !equality.Semantic.DeepEqual(existing.Spec, required.Spec) {
-		*modified = true
-		existing.Spec = required.Spec
+// ensureCustomResourceDefinitionV1CaBundle ensures that the field
+// spec.Conversion.Webhook.ClientConfig.CABundle of a CRD is not managed by the CVO when
+// the service-ca controller is responsible for the field.
+func ensureCustomResourceDefinitionV1CaBundle(required *apiextv1.CustomResourceDefinition, existing apiextv1.CustomResourceDefinition) {
+	if val, ok := existing.ObjectMeta.Annotations[injectCABundleAnnotation]; !ok || val != "true" {
+		return
+	}
+	if required.Spec.Conversion != nil && required.Spec.Conversion.Strategy == apiextv1.WebhookConverter {
+		// The service CA bundle is only injected by the service-ca controller into
+		// the CRD if the CRD is configured to use a webhook for conversion
+		if required.Spec.Conversion.Webhook != nil &&
+			required.Spec.Conversion.Webhook.ClientConfig != nil {
+			if existing.Spec.Conversion != nil &&
+				existing.Spec.Conversion.Webhook != nil &&
+				existing.Spec.Conversion.Webhook.ClientConfig != nil {
+				required.Spec.Conversion.Webhook.ClientConfig.CABundle = existing.Spec.Conversion.Webhook.ClientConfig.CABundle
+			}
+		}
```

Comment:
```suggestion
	req := required.Spec.Conversion
	if req == nil ||
		req.Strategy != apiextv1.WebhookConverter ||
		req.Webhook == nil ||
		req.Webhook.ClientConfig == nil {
		return
	}
	// The service CA bundle is only injected by the service-ca controller into
	// the CRD if the CRD is configured to use a webhook for conversion
	if exc := existing.Spec.Conversion; exc != nil && exc.Webhook != nil && exc.Webhook.ClientConfig != nil {
		required.Spec.Conversion.Webhook.ClientConfig.CABundle = exc.Webhook.ClientConfig.CABundle
	}
```

I'm sorry for bikeshedding style this late in the PR but I think we can simplify for readability with guardclauses and some variable extraction... I don't insist, just consider.

Reply:
Don't be sorry! Thank you for the suggestions. I appreciate them.

What do you say about something like this? I tried to make it even more consistent (added a new line for `exc := existing.Spec.Conversion`) and I tried to separate the logic even more (added a new condition only including the `WebhookConverter` check).

```go
	req := required.Spec.Conversion
	if req == nil ||
		req.Webhook == nil ||
		req.Webhook.ClientConfig == nil {
		return
	}
	if req.Strategy != apiextv1.WebhookConverter {
		// The service CA bundle is only injected by the service-ca controller into
		// the CRD if the CRD is configured to use a webhook for conversion
		return
	}
	exc := existing.Spec.Conversion
	if exc != nil &&
		exc.Webhook != nil &&
		exc.Webhook.ClientConfig != nil {
		required.Spec.Conversion.Webhook.ClientConfig.CABundle = exc.Webhook.ClientConfig.CABundle
	}
```

Reply:
that's good. personally I wouldn't linebreak the conditions in the `if` statements, the resulting lines are short. but really that's just nitpicking. I think the main benefit comes from extracting the long dot-chains to variables :+1:

Reply:
The new structure looks good either way.

Reply:
I can add a lgtm once the changes are done.

Reply:
I have pushed the above mentioned solution by me (https://github.com/openshift/cluster-version-operator/pull/870#discussion_r1081545493) and changed the line:

```go
		required.Spec.Conversion.Webhook.ClientConfig.CABundle = exc.Webhook.ClientConfig.CABundle
```

to:

```go
		req.Webhook.ClientConfig.CABundle = exc.Webhook.ClientConfig.CABundle
```

Because the variable `req` is a pointer `*apiextv1.CustomResourceConversion` and assigning the `CABundle` does change the value in the original variable.